spyder/app/tests/test_mainwindow.py

```diff
@@ -679,5 +679,24 @@ def test_varexp_magic_dbg(main_window, qtbot):
     assert shell._control.toHtml().count('img src') == 1
 
 
+@flaky(max_runs=3)
+def test_change_cwd_ipython_console(main_window, qtbot, tmpdir):
+    """
+    Test synchronization between working directoy ipython console and explorer.
+    """
+    # Wait until the window is fully up
+    shell = main_window.ipyconsole.get_current_shellwidget()
+    qtbot.waitUntil(lambda: shell._prompt_html is not None, timeout=SHELL_TIMEOUT)
+
+    tmpdirname = tmpdir.mkdir("test_dir")
+
+    with qtbot.waitSignal(shell.executed):
+        shell.execute("%cd {}".format(tmpdirname))
+
+    qtbot.wait(1000)
+
+    assert main_window.workingdirectory.history[-1] == str(tmpdirname)
```

Comment:
Also include in this test that changing the cwd in the File Explorer updates the cwd in our IPython consoles.